### PR TITLE
Add InfuxDB support in export distro

### PR DESCRIPTION
**Pull request by davidmsh**
- Change the Send function of Sender interface to pass in the original event as we don't need the formatted message but the readings in the event

- Add a "NOOP" formatter which, as its name suggests, does nothing. This formatter is used when we do not used the formatted form of the event

- Add a new "INFLUXDB_ENDPOINT" destination type and implement NewInfluxDBSender
to handle it by sending the reading to influxdb

### CODE REVIEW

File: cmd/core-metadata/res/configuration.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 DBType = 'mongodb'
 MongoDatabaseName = 'metadata'
-MongoDBUserName = ''
-MongoDBPassword = ''
+MongoDBUserName = 'meta'
```
Comment (drasko) [inline cmd/core-metadata/res/configuration.toml]: Why is this needed? Why not leaving empty strings?

File: cmd/export-client/res/configuration.toml [hunk 2]
```diff
@@ -2,8 +2,8 @@ Hostname = '127.0.0.1'
 Port = 48071
 DBType = 'mongodb'
 MongoURL = '0.0.0.0'
-MongoUsername = ''
-MongoPassword = ''
+MongoUsername = 'core'
+MongoPassword = 'password'
 MongoDatabase = 'coredata'
```
Comment (drasko) [inline cmd/export-client/res/configuration.toml]: Looks like in other `toml` we are using `MongoDatabaseName`. Please align.

File: export/distro/format.go [hunk 3]
```diff
@@ -167,3 +168,11 @@ func (af azureFormatter) Format(event *models.Event) []byte {
 	}
 	return msg
 }
+
+type noopFormatter struct {
+}
+
+func (noopTr noopFormatter) Format(event *models.Event) []byte {
+	var ret []byte
+	return ret
```
Comment (drasko) [inline export/distro/format.go]: No need for declaring a var. You can `ret []byte` directly.

BTW. why is this a separate function? When you need empty byte array in the code, can't you just declare the one over there?

File: export/distro/format.go [hunk 4]
```diff
@@ -167,3 +168,11 @@ func (af azureFormatter) Format(event *models.Event) []byte {
 	}
 	return msg
 }
+
+type noopFormatter struct {
```
Comment (drasko) [inline export/distro/format.go]: I do not see use in declaring an empty struct as a separate type...

File: export/distro/influxdb.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	httpInfo client.HTTPConfig
+	database string
+}
+
+func NewInfluxDBSender(addr models.Addressable) Sender {
+	connStr := "http://" + addr.Address + ":" + strconv.Itoa(addr.Port)
+
+	influxdbHTTPInfo := client.HTTPConfig{
+		Addr:     connStr,
+		Timeout:  time.Duration(60000) * time.Millisecond,
```
Comment (drasko) [inline export/distro/influxdb.go]: Do not hardcode numeric constants. No magic numbers, please declare a const.

File: export/distro/format.go (hunk 4 above)
Comment (SteveOss) [reply export/distro/format.go]: Slightly disagree here. The type name does impart some additional information i.e the intended usage. In C/C++ quite often see string types, typedef'd to something more meaningful.

File: export/distro/format.go (hunk 3 above)
Comment (feclare) [reply export/distro/format.go]: @drasko They just want to create a type that satisfies the ```Formatter``` interface. It does nothing because they only use the ```Sender``` interface of the distro pipeline. 
How do you recommend to do that without defining a type?

**Comment by feclare** [comment]
@davidmsh Tests need to be fixed. 

Please run ```make test``` before creating a PR.

File: export/distro/format.go (hunk 4 above)
Comment (drasko) [reply export/distro/format.go]: Especially C++ produces a lot of bloat that go philosophy tries to avoid. Feel free to return empty arrays or strings directly from the code, leave a short comment if necessary, but oplease do not bloat the code.

Every code line is potential bug introduced, keep them at minimum.

File: export/distro/format.go (hunk 3 above)
Comment (davidmsh) [reply export/distro/format.go]: Looks like "return []byte" is not legal, I will change it to "return []byte{}"
As @feclare said, the separate function is to match the Formatter interface we already have for formatter. I return a byte array to make the function in the interface. It's empty because this formatter is meant to do nothing, that why it's called noop.

File: export/distro/format.go (hunk 4 above)
Comment (davidmsh) [reply export/distro/format.go]: The empty struct looked strange to me when I first saw the code as a new golang developer. But it's pretty much like a class to implement certain interface, just golang does require you list all the function in the interface in the implementing class.

File: export/distro/influxdb.go (hunk 5 above)
Comment (davidmsh) [reply export/distro/influxdb.go]: Will do.

File: cmd/core-metadata/res/configuration.toml (hunk 1 above)
Comment (davidmsh) [reply cmd/core-metadata/res/configuration.toml]: I thought it's more convenient to leave some default values here. I can change it back.

**Comment by davidmsh** [comment]
@drasko I have addressed your requested changes. Can you check again if you are ok with it ?

**Comment by feclare** [comment]
@davidmsh Can you add a test for the added formatter in format_test.go? Just test that it outputs empty arrays.

**Comment by davidmsh** [comment]
@feclare Done adding the test case for noop formatter

**Comment by drasko** [comment]
@feclare and @tsconn23 please review this one, because it introduces `event` in `Send()` signature.

**Comment by feclare** [comment]
Who is updating the configurations files in config-seed? Because there is a field change in this PR.

**Comment by davidmsh** [comment]
@feclare You suggested to align the name MongoDatabase with others as MongoDatabaseName.

**Comment by davidmsh** [comment]
@feclare Sorry, it's @drasko who made such suggestion

**Comment by tsconn23** [comment]
@davidmsh Please rebase and commit your branch to clear the "out of date" message above then we can pull this in.
